waspc/cli/src/Wasp/Cli/Command/Build.hs

```diff
@@ -58,40 +73,65 @@ build = do
     throwError $
       CommandError "Building of wasp project failed" $ show (length errors) ++ " errors found."
 
-  liftIO $ copyUserFilesNecessaryForBuild waspProjectDir buildDir
-
-  cliSendMessageC $
-    Msg.Success "Your wasp project has been successfully built! Check it out in the .wasp/build directory."
+  liftIO (prepareFilesNecessaryForDockerBuild waspProjectDir buildDir) >>= \case
+    Left err -> throwError $ CommandError "Failed to prepare files necessary for docker build" err
+    Right () ->
+      cliSendMessageC $
+        Msg.Success "Your wasp project has been successfully built! Check it out in the .wasp/build directory."
```

Comment:
```suggestion
  liftIO (prepareFilesNecessaryForDockerBuild waspProjectDir buildDir) >>= \case
    Left err -> throwError $ CommandError "Failed to prepare files necessary for docker build" err
    Right () -> return ()
  cliSendMessageC $
    Msg.Success "Your wasp project has been successfully built! Check it out in the .wasp/build directory."
```
Maybe pull out the success msg so in the future we can add more steps before without a rewrite?

Reply:
Sure, that's what we had before. I'll change it back.

Reply:
`unless` was ugly because I'd have to force `fromLeft` with a bogus return value, so I went with @infomiho's suggestion.